src/hooks/useIsInView.ts

```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from 'react';
+
+export const useIsInView = (ref: React.RefObject<HTMLElement | null>): boolean => {
+  const [isInView, setIsInView] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!('IntersectionObserver' in window)) {
```

Comment:
We could remove the IntersectionObeserver once "isInView" is true, to improve performance - although that might be a different hook "useOnceInView()" or something.